crates/brioche-core/src/download.rs

```diff
@@ -41,7 +41,7 @@ pub async fn download(
         }
     });
 
-    let mut download_stream = response
+    let download_stream = response
```

Comment:
`download_stream` gets wrapped with `std::pin::pin!()`. In Rust v1.88, this started triggering an `unused_mut` warning.

I couldn't find anything in the release notes that would explain why this is now a warning, but I'm guessing it's probably a result of this change: https://github.com/rust-lang/rust/pull/139114